### PR TITLE
ci: Enable conventional commits

### PR DESCRIPTION
**Pull request by hoxbro**
Will merge it after the next release

### CODE REVIEW

File: .github/workflows/conventional_commits.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+          types: |
+            build
+            chore
+            ci
+            compat
+            docs
+            enh
+            feat
+            fix
+            perf
```
Comment (hoxbro) [inline .github/workflows/conventional_commits.yaml:32]:
```suggestion
            perf
            refactor
```

**Review by maximlt — COMMENTED** [review]
Does the contributor guide need to be updated?

**Comment by hoxbro** [comment]
> Does the contributor guide need to be updated?

Done in https://github.com/holoviz/holoviews/pull/6397/commits/cf0ce35fc4d1d24859212683356e6f3c6e8d7256

File: doc/developer_guide/index.md [hunk 2]
```diff
@@ -181,6 +181,35 @@ pixi run build-pip
...
+        * docs: Documentation only changes
+        * enh: An enhancement to existing feature
+        * feat: A new feature
+        * fix: A bug fix
+        * perf: A code change that improves performance
+        * refactor: A code change that neither fixes a bug nor adds a feature
+        * test: Adding missing tests or correcting existing tests
+        * type: Type annotations
+
+1.  Write a description of your changes in the.
```
Comment (maximlt) [inline doc/developer_guide/index.md:210]: This sentence looks incomplete.

Comment (hoxbro) [reply doc/developer_guide/index.md:210]:
``` suggestion
1.  Write a description of your changes in the `Write` tab, and check if everything looks ok in the `Preview` tab.
```

**Pull request merged**